Sources/Concordium/Domain/Transaction.swift

```diff
@@ -197,121 +266,423 @@ public struct SignedAccountTransaction: ToGRPC {
+        }
+        if let transactionFeeCommission = transactionFeeCommission {
+            res += buffer.writeSerializable(transactionFeeCommission)
+        }
+        if let bakingRewardCommission = bakingRewardCommission {
+            res += buffer.writeSerializable(bakingRewardCommission)
+        }
+        if let finalizationRewardCommission = finalizationRewardCommission {
+            res += buffer.writeSerializable(finalizationRewardCommission)
+        }
+        return res
+    }
+}
+
+public struct ConfigureDelegationPayload: Equatable, Serialize, Deserialize {
+    /// The equity capital of the baker
+    public let capital: CCD?
+    /// Whether the baker's earnings are restaked
+    public let restakeEarnings: Bool?
+    /// The delegation target to add the stake to
+    public let delegationTarget: DelegationTarget?
+
+    public init(capital: CCD? = nil, restakeEarnings: Bool? = nil, delegationTarget: DelegationTarget? = nil) {
+        self.capital = capital
+        self.restakeEarnings = restakeEarnings
+        self.delegationTarget = delegationTarget
+    }
+
+    public func serializeInto(buffer: inout NIOCore.ByteBuffer) -> Int {
+        var res = 0
+
+        var bitmap: UInt16 = 0
+        func setBit(_ pos: Int, cond: Bool) {
+            if cond { bitmap = bitmap | (1 << pos) }
+        }
+
+        setBit(0, cond: capital != nil)
+        setBit(1, cond: restakeEarnings != nil)
+        setBit(2, cond: delegationTarget != nil)
+        res += buffer.writeInteger(bitmap)
+
+        if let capital = capital {
+            res += buffer.writeSerializable(capital)
+        }
+        if let restakeEarnings = restakeEarnings {
+            res += buffer.writeBool(restakeEarnings)
+        }
+        if let delegationTarget = delegationTarget {
+            res += buffer.writeSerializable(delegationTarget)
+        }
+
+        return res
+    }
+
+    public static func deserialize(_ data: inout Cursor) -> ConfigureDelegationPayload? {
+        var capital: CCD?
+        var restakeEarnings: Bool?
+        var delegationTarget: DelegationTarget?
+
+        guard let bitmap = data.parseUInt(UInt16.self) else { return nil }
+        if bitmap & 1 == 1 {
+            guard let amount = CCD.deserialize(&data) else { return nil }
+            capital = amount
+        }
+        if bitmap & (1 << 1) != 0 {
+            guard let restake = data.parseBool() else { return nil }
+            restakeEarnings = restake
+        }
+        if bitmap & (1 << 2) != 0 {
+            guard let target = DelegationTarget.deserialize(&data) else { return nil }
+            delegationTarget = target
+        }
+
+        return .init(capital: capital, restakeEarnings: restakeEarnings, delegationTarget: delegationTarget)
+    }
+}
+
+/// The payload for an account transaction. Only contains payloads valid from protocol version 7
 public enum AccountTransactionPayload: Serialize, Deserialize, FromGRPC, ToGRPC, Equatable {
```

Comment:
@ramakser One thing to node is that this type does not conform to the `Codable` protocol.

You'll receive transactions through wallet-connect in JSON format, so I'm guessing you'd have to be able to decode the JSON into this?

Reply:
If this is needed, I'll have to do that in a subsequent PR.

Reply:
I think that's okay for now. Anyway we will have separate model on mobile side to decode data from json string

Reply:
I think it would make sense if you could use the models added in the SDK though so you could get rid of the models in the mobile application. The point of the SDK is also to enable third parties in creating a wallet, so it would be nice if our own wallet would follow as closely as possible the approach a third party would take to implementing a wallet.

I think we need to add `Codable` to the `AccountTransactionPayload` type to enable this, but as previously stated this will come in a subsequent PR.